server/asset/btc/btc.go

```diff
@@ -24,6 +24,18 @@ import (
 	"github.com/btcsuite/btcutil"
 )
 
+// Driver implements asset.Driver.
+type Driver struct{}
+
+// Start creates the BTC backend and verifies the chain server connection.
+func (d *Driver) Start(ctx context.Context, configPath string, logger dex.Logger, network dex.Network) (asset.DEXAsset, error) {
+	return NewBackend(ctx, configPath, logger, network)
+}
+
+func init() {
+	asset.Register(assetName, &Driver{})
+}
```

Comment:
That's a good suggestion.  The main reason is because NewBackend in the `dcr` package does not return a `asset.DEXAsset` but a `*dcr.Backend`.  As for `btc` (and clones), that's a good point.

Reply:
One reason I found to register a `Driver` type instead of a function is that the driver might want to do other things.  For example, in the DB driver I added a UseLogger method that calls the packages UseLogger, which allowed consumers of server/db to provide the logger to the driver's package which wouldn't be appropriate for the constructor since it's package-level and not per-instance.  Maybe there's better ways of dealing with that logging issue, but having the driver defined was handy there.
But regarding dcr.NewBackend returning a `*dcr.DCRBackend`, wrapping the constructor in that case was necessary anyway so I'm fairly content with the Driver struct at the moment.
OK if we run with this and think on it?

Reply:
Absolutely.